File: backend/app/schemas/wolistic.py
"""Pydantic schemas for AI / Wolistic search responses."""

from __future__ import annotations

import uuid

from pydantic import BaseModel

from app.schemas.professional import ProfessionalProfileOut


class ProductOut(BaseModel):
    id: uuid.UUID
    name: str
    image: str | None = None
    category: str | None = None
    brand: str | None = None
    website_name: str | None = None
    website_url: str | None = None
    description: str | None = None
    price: int = 0


class WolisticServiceOut(BaseModel):
    id: uuid.UUID
    title: str
    type: str
    location: str
    image_url: str | None = None
    website_name: str | None = None
    website_url: str | None = None
    tags: list[str] = []


class WolisticArticleOut(BaseModel):
    id: uuid.UUID
    title: str
    read_time: str


class WolisticSearchResponse(BaseModel):
    professionals: list[ProfessionalProfileOut] = []
    products: list[ProductOut] = []
    services: list[WolisticServiceOut] = []
    articles: list[WolisticArticleOut] = []

File: frontend/types/wolistic.ts
export type Product = {
  id: string;
  name: string;
  image?: string;
  category?: string;
  brand?: string;
  websiteName?: string;
  websiteUrl?: string;
  description?: string;
  price: number;
};

export type WolisticService = {
  id: string;
  title: string;
  type: string;
  location: string;
  imageUrl?: string;
  websiteName?: string;
  websiteUrl?: string;
  tags: string[];
};

export type WolisticArticle = {
  id: string;
  title: string;
  readTime: string;
};

export type WolisticSearchResult = {
  professionals: import("./professional").ProfessionalProfile[];
  products: Product[];
  services: WolisticService[];
  articles: WolisticArticle[];
};
